clients/pubsub-common/src/main/java/org/eclipse/hono/client/pubsub/PubSubBasedAdminClientManager.java

```diff
@@ -171,37 +169,23 @@ private Future<Void> getOrCreateTopic(final TopicName topic) {
     }
 
     private Future<Void> getOrCreateTopic(final String projectId, final TopicName topic) {
-        if (topicExists(topic.toString())) {
+        if (topics.contains(topic.toString())) {
+            LOG.debug("Topic {} already exists, continue", topic);
             return Future.succeededFuture();
         }
-        final Topic createdTopic = topicAdminClient.createTopic(topic);
-        if (createdTopic == null) {
-            LOG.debug("Creating topic failed [topic: {}, projectId: {}]", topic.toString(), projectId);
-            return Future.failedFuture("Topic creation failed.");
-        }
-        topics.add(topic.toString());
-        return Future.succeededFuture();
-    }
-
-    private boolean topicExists(final String topicId) {
-        if (topics.contains(topicId)) {
-            return true;
-        }
-        final ProjectName projectName = ProjectName.of(projectId);
         try {
-            final ListTopicsPagedResponse pagedResponse = topicAdminClient.listTopics(projectName);
-            if (pagedResponse != null) {
-                for (Topic topic : pagedResponse.iterateAll()) {
-                    if (topic.getName().equals(topicId)) {
-                        topics.add(topicId);
-                        return true;
-                    }
-                }
+            final Topic createdTopic = topicAdminClient.createTopic(topic);
+            if (createdTopic == null) {
+                LOG.error("Creating topic failed [topic: {}, projectId: {}]", topic, projectId);
+                return Future.failedFuture("Topic creation failed.");
             }
-            return false;
+            topics.add(topic.toString());
+            return Future.succeededFuture();
+        } catch (AlreadyExistsException ex) {
+            return Future.succeededFuture();
         } catch (ApiException e) {
-            LOG.error("Error listing topics on project {}", projectId, e);
-            return false;
+            LOG.error("Error creating topic {} on project {}", topic, projectId, e);
```

Comment:
I wonder if the *error* level is actually appropriate for logging the failures in this class. In the end, we are handling all these issues properly and the overall system does not run into an ERROR state from which it needs to be recovered (manually), right?
FMPOV a level of DEBUG or maybe INFO would be sufficient here ...

Reply:
Agree, updated them to log level DEBUG